include/swift/module.modulemap

```diff
@@ -0,0 +1,157 @@
+    textual header "AST/DiagnosticsSema.def"
+    textual header "AST/ExprNodes.def"
+    textual header "AST/KnownDecls.def"
+    textual header "AST/KnownFoundationEntities.def"
+    textual header "AST/KnownIdentifiers.def"
+    textual header "AST/KnownProtocols.def"
+    textual header "AST/KnownStdlibTypes.def"
+    textual header "AST/NameLookupTypeIDZone.def"
+    textual header "AST/PatternNodes.def"
+    textual header "AST/PlatformKinds.def"
+
+    // Missing include.
+    textual header "AST/DiagnosticsCommon.h"
+    textual header "AST/FileSystem.h"
+
+    // Included inside a namespace.
+    textual header "Basic/DefineTypeIDZone.h"
+    textual header "Basic/ImplementTypeIDZone.h"
+  }
+
+  module Basic {
+    requires cplusplus
+    umbrella "Basic" module * { export * }
+    textual header "Basic/CTypeIDZone.def"
+    textual header "Basic/FileTypes.def"
+    textual header "Basic/Statistics.def"
+
+    // This is included by Swift_Shims.
+    textual header "Basic/type_traits.h"
+    textual header "Basic/Compiler.h"
+  }
+
+  module Demangling {
+    requires cplusplus
+    umbrella "Demangling" module * { export * }
+    textual header "Demangling/DemangleNodes.def"
+    textual header "Demangling/StandardTypesMangling.def"
+    textual header "Demangling/ValueWitnessMangling.def"
+  }
+
+  module Frontend {
+    requires cplusplus
+    umbrella "Frontend" module * { export * }
+  }
+
+  module Parse {
+    requires cplusplus
+    umbrella "Parse" module * { export * }
+  }
+
+  module IDE {
+    requires cplusplus
+    umbrella "IDE" module * { export * }
+    textual header "IDE/DigesterEnums.def"
+    textual header "IDE/RefactoringKinds.def"
+  }    
+
+  module Reflection {
+    requires cplusplus
+    umbrella "Reflection" module * { export * }
+    textual header "Reflection/MetadataSources.def"
+    textual header "Reflection/TypeRefs.def"    
+  }
+
+  module Remote {
+    requires cplusplus
+    umbrella "Remote" module * { export * }
+    textual header "Remote/FailureKinds.def"
+  }
+
+  module RemoteAST {
+    requires cplusplus
+    umbrella "RemoteAST" module * { export * }
+  }
+
+  module Runtime {
+    requires cplusplus
+    umbrella "Runtime" module * { export * }
+   
+    // These two headers introduce a cyclic dependency with Swift_Shims.
+    textual header "Runtime/Config.h"
+    textual header "Runtime/HeapObject.h"
+   
+    textual header "Runtime/BuiltinTypes.def"
+
+    textual header "Runtime/Atomic.h"
+    textual header "Runtime/Debug.h"
+    textual header "Runtime/Unreachable.h"
+   
+    module MutexWin32 {
+      header "Runtime/MutexWin32.h"
+      requires windows
+    }
+  }
+
+  module SIL {
+    requires cplusplus
+    umbrella "SIL" module * { export * }
+    textual header "SIL/AccessedStorage.def"
+    textual header "SIL/BridgedTypes.def"
+    textual header "SIL/SILNodes.def"
+  }
+
+  extern module SwiftShims "../../stdlib/public/SwiftShims/module.modulemap"
```

Comment:
Um. This part's unfortunate. SwiftShims can't just depend on Swift.Runtime and Swift.ABI?